persistent/persistence.py

```diff
@@ -95,12 +98,18 @@ def _get_oid(self):
     def _set_oid(self, value):
         if value == self.__oid:
             return
-        if value is not None:
-            if not isinstance(value, OID_TYPE):
-                raise ValueError('Invalid OID type: %s' % value)
-        if self.__jar is not None and self.__oid is not None:
-            raise ValueError('Already assigned an OID by our jar')
-        self.__oid = value
+        # The C implementation allows *any* value to be
+        # used as the _p_oid.
```

Comment:
I suppose it is a slight convenience in certain test cases to be able to assign them arbitrary values, especially if you're worried about supporting different Python versions. Compare `o._p_oid = 42` to `from ._compat import _b; o._p_oid = _b('abc')`.

From a correctness standpoint, though, not only did the C implementation accept *any* value for `_p_oid`, both implementations accepted *any* byte string; that is, neither one of them confirmed that the value actually had a length of exactly 8. I could be said that a half-way check is not of much use and is arguably worse than no check at all. And a full (or even half) check is a breaking change; how much of one, I don't know.